### ISSUE TITLE
Add stacked lines to stacked area plot

### ISSUE THREAD

**Issue by harabat** [bokeh/bokeh#10047]
**Is your feature request related to a problem? Please describe.**
I wonder whether stacked lines could be added to stacked area plots (`vstack_area()`), which is the default in Seaborn or Plotly for example. 

**Describe the solution you'd like**
Stacked line (`vline_stack()`) and area (`vstack_area()`) plots take the same arguments. One solution would be to just call `vline_stack` every time `vstack_area` is called. The solution I wanted to suggest in a PR for discussion is a bit more flexible, as it allows a `line_width` argument to be specified, in which case `vline_stack` is called. 
`bokeh/plotting/figure.py`:
```
def varea_stack(self, stackers, **kw):
    result = []
    if 'line_width' in kw:
        line_width = kw['line_width']
        kw.pop('line_width')
        result.append(self.vline_stack(stackers=stackers, line_width=line_width, **kw))
    for kw in double_stack(stackers, "y1", "y2", **kw):
        result.append(self.varea(**kw))
    return result
```
instead of 
```
def varea_stack(self, stackers, **kw):
    for kw in double_stack(stackers, "y1", "y2", **kw):
        result.append(self.varea(**kw))
    return result
```
This relies on `vline_stack()` and `vstack_area()` not diverging in future versions however. You will also notice that the legend has changed: it can be fixed if people think it's an issue. 

**Describe alternatives you've considered**
One could just call `vline_stack()` whenever they make a stacked area plot, but other libraries seem to have chosen a different default. 

**Additional context**
Without stacked lines
![Without stacked lines][1]

With stacked lines
![With stacked lines][2]

[1]: https://i.stack.imgur.com/DjVM9.png
[2]: https://i.stack.imgur.com/JlTnX.png

**Comment by bryevdv** [comment]
@harabat We could possibly consider something like this, but the code above would need to be made more robust:

* Currently, things work by coincidence because you pass the helper alias `alpha`. But if you passed the more specific `fill_alpha` then the example above would raise and error, since line glyphs do not have a `fill_alpha`
* It would also behave unexpectedly if someone passed `line_color` or `line_dash` instead of `line_width` (because they assume  it would default to a value of 1, for example)

Basically, the new function code would need to carefully and completely split out *all* line properties and fill properties from kwargs up front, and pass only what is explicitly appropriate to each of `varea` and `vline_stack`

cc @bokeh/dev thoughts?

**Comment by harabat** [comment]
Sorry, I didn't spend the time I should have on this, my solution is very limited. Having to open an issue before doing a PR is definitely good practice. 

Thinking about it more, I came to view stacked area charts and stacked line plots as one and the same: the default is lines, fill properties are options. I looked into other libs:
- Matplotlib has a single [`plt.stackplot()`](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/stackplot.py) method, lines are invisible even with a `linewidth` argument: there is no function to draw stacked lines with no fill or to specify no fill to `stackplot()`
![plt.stackplot()][0]
- importing Seaborn makes `plt.stackplot()` return stacked plots with white lines by default, no need to specify line_width, but there is no method to make stacked lines with no fill:
![stackplot()][1]
- Pandas' `pd.DataFrame.plot.area()` will return a stacked plot with lines (they're barely visible without specifying `linewidth` or `alpha` though), no method to make stacked lines with no fill:
![pd.DataFrame.plot.area()][2]
- in Plotly, everything goes into `go.Scatter()` with a `stackgroup` argument specified: a stacked line chart is a stacked area plot with `fill='none'`, while the default plot will be a stacked area with stacked lines:
![go.Scatter(stacked='stackgroup')][3]

In every case above with lines, lines have a slightly higher alpha than the areas, or at least appear to be slightly more opaque. 

In the context of this discussion, I'd suggest the following:
- I think a `line_width` argument could be added to `class VArea`, with colour being the same as the areas' one and alpha being slightly higher, like in other libs, as I believe specifying allowing to specify `line_alpha` and `line_color` that are different from the areas' ones would make it more messy without being a popular feature
- otherwise, `_line_stack()` and `varea_stack()` could maybe be merged by adding `line_props` to `class VArea` in `models/glyphs.py` (and everything else that this entails)

[0]: https://i.stack.imgur.com/eJ71M.png
[1]: https://i.stack.imgur.com/592J2.png
[2]: https://i.stack.imgur.com/1WDVQ.png
[3]: https://i.stack.imgur.com/DnJJP.png

**Comment by bryevdv** [comment]
@harabat Offhand I think I am -1 on adding line properties to the low level `VArea` itself. I think your initial implementation, that confines changes to the higher level "stacked" glyph helpers, is a perfectly reasonable direction to pursue, it just needs to be more careful and thorough about handling the arguments. I'll have to think about it some more. 

Regardless, I am definitely reticent to start defaulting the alpha to anything other than 1. There is nowhere in Bokeh that this is the case, and apart from being a breaking change, I think it is good to preserve the consistent expectation "If I want alpha, I have to set it". I'd be more inclined to follow the seaborn example where the line and area alpha are both 1 but the line color defaults to white to leave "line gaps" between the areas. We would want to start off with a default `line_color=None` in 2.x to avoid breaking changes, then change to `line_color="white"` in Bokeh 3.0 when a breaking change can be made.

**Comment by bryevdv** [comment]
Another option is to leave the existing functions alone, and make an entirely new helper that coordinates coordinates the "plain" `varea_stack` and `vline_stack`, e.g. 
```
plot.vstack(...)
```

@birdsarah @hyles-lineata would love to know if y'all have any thoughts/preferences here

**Comment by harabat** [comment]
Thanks for spending time on this @bryevdv, your solution is way simpler to implement, happy to wait and see what others think. My only issue with my initial solution was that it relied on common evolution of varea_stack() and vline_stack(), but if you think it's appropriate, I'm down.

**Comment by bryevdv** [comment]
> it relied on common evolution of varea_stack() and vline_stack()

I think this is an overly pessimistic assessment. I think it can be assumed that: `VArea` will always take `FillProps` and `VLine` will always take `LineProps`, and, even if the definitions of `FillProps` and `LineProps` were to change, there are generic ways to extract `FillProps` and `LineProps` from `kwargs`, so those changes would be accommodated automatically.

**Comment by harabat** [comment]
I was talking about the key arguments, like `stackers`, not the cosmetic ones, but your statement also holds for these I assume.

**Comment by birdsarah** [comment]
I haven't used this API in a long time so I don't have any deep opinions. I like the white line idea. I'm not sure about stacked vs the varea vline plan, so other people's judgement is welcome.

**Comment by bryevdv** [comment]
Correct I would not ever expect those to change

**Comment by harabat** [comment]
I assume we're still waiting for input about this feature. In the meantime, I found a discrepancy between `varea_stack()` and `vline_stack()` for the `stackers` argument, made an issue about it: #10067

**Comment by harabat** [comment]
We're probably not getting any more opinions about this, should I go forward and make a PR?

**Comment by bryevdv** [comment]
@harabat which approach did you intend to pursue?

**Comment by harabat** [comment]
Turns out I forgot we had identified several options and didn't actually decide on either.

Rereading through the discussions, I'm more inclined to go with my original approach: confine changes to `varea_stack`, while making sure all `lineprops` arguments are accounted for, and add line renderers with `line_color=None` by default (to switch to `line_color='white'` in 3.0). 

My gripes with the option of adding a new vstack helper are that it'd be a confusing stack function, compared with the more clear cut `varea_stack`, `vline_stack`, and `vbar_stack`. Also, I can't help but side with Seaborn and Pandas on the idea of having lines by default. 

Obviously, the above holds for the horizontal equivalents.
